### PR TITLE
Remove .spec.volumes from role hash for pods

### LINKED ISSUE #1697: Kueue assumes Pods with different names of volumes belong to different pod groups
**What happened**:

Created 9 pods in a pod group of size 8. Kueue failed to a create the Workload object because it thought each pod was a different pod set. Kueue appears to have a max pod set size of 8: https://github.com/kubernetes-sigs/kueue/blob/892d85c35a94c875005d6b547710ad06704bccf9/pkg/controller/jobs/pod/pod_controller.go#L885

Each pod had a different role hash. It appears that this is due to each pod having a different PVC name specified in `.spec.volumes`.

**What you expected to happen**:

I would expect Kueue to ignore PVC volume names or volumes altogether and treat the 9 pods as having the same role (being in the same pod set). See: https://github.com/kubernetes-sigs/kueue/blob/892d85c35a94c875005d6b547710ad06704bccf9/pkg/controller/jobs/pod/pod_controller.go#L565

**How to reproduce it (as minimally and precisely as possible)**:

Follow the steps above, using a different PVC name in the volumes array for each pod. Make sure to use N > 8 for pod group size. Make sure Kueue is configured to support raw pods. Assert that the Workload never gets created. The pods should remain in a ScheduleGated state.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): NA
- Kueue version (use `git describe --tags --dirty --always`): v0.6.0-rc.1
- Cloud provider or hardware configuration: NA
- OS (e.g: `cat /etc/os-release`): A
- Kernel (e.g. `uname -a`): NA
- Install tools: NA
- Others: NA

### PR DESCRIPTION
**Pull request by nstogner**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

Removes `.spec.volumes` from pod role hash calculation. Allows for users to specify different volumes for pods in a pod group.

#### Which issue(s) this PR fixes:

Fixes #1697 

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: pkg/controller/jobs/pod/pod_controller.go [hunk 1]
```diff
@@ -562,7 +562,6 @@ func getRoleHash(p corev1.Pod) (string, error) {
 			"preemptionPolicy":          p.Spec.PreemptionPolicy,
 			"topologySpreadConstraints": p.Spec.TopologySpreadConstraints,
 			"overhead":                  p.Spec.Overhead,
-			"volumes":                   volumesShape(p.Spec.Volumes),
```
Comment (alculquicondor) [inline pkg/controller/jobs/pod/pod_controller.go:565]: remove the code for the function.

Comment (nstogner) [reply pkg/controller/jobs/pod/pod_controller.go:565]: Duh - whoops, doing that now

Comment (nstogner) [reply pkg/controller/jobs/pod/pod_controller.go:565]: Done

**Review by alculquicondor — COMMENTED** [review]
/lgtm

**Pull request merged**